src/app/business/delfin/storages/storages.component.ts

```diff
@@ -0,0 +1,601 @@
+import { Component, OnInit, ViewContainerRef, ViewChild, Directive, ElementRef, HostBinding, HostListener } from '@angular/core';
+import { Router } from '@angular/router';
+import { I18NService, Utils } from 'app/shared/api';
+import { FormControl, FormGroup, FormBuilder, Validators, ValidatorFn, AbstractControl } from '@angular/forms';
+import { AppService } from 'app/app.service';
+import { I18nPluralPipe } from '@angular/common';
+import { trigger, state, style, transition, animate } from '@angular/animations';
+import { Message, MenuItem ,ConfirmationService} from '../../../components/common/api';
+import { DelfinService } from '../../delfin/delfin.service';
+import { TreeNode } from '../../../components/common/api';
+import { isArray } from 'underscore';
+import { OverlayPanel } from 'app/components/overlaypanel/overlaypanel';
+
+let _ = require("underscore");
+@Component({
+    selector: 'app-delfin-storages',
+    templateUrl: 'storages.component.html',
+    providers: [ConfirmationService],
+    styleUrls: [],
+    animations: []
+})
+export class StoragesComponent implements OnInit {
+    allArrays: TreeNode[];
+    arrayTreeData: TreeNode[];
+    allStorages: any = [];
+    selectedStorages: any = [];
+    selectedStorageId: any;
+    selectStorage;
+    showListView: boolean = false;
+    menuItems: MenuItem[];
+    capacityData: any;
+    chartOptions: any;
+    vendorOptions;
+    modelOptions;
+    allStorageModels;
+    versionOptions: any;
+    selectedVersion: any;
+    securityLeveloptions: any;
+    selectedSecurityLevel: any;
+    authProtocolOptions: any;
+    selectedAuthProtocol: any;
+    privacyProtocolOptions: any;
+    selectedPrivacyProtocol: any;
+    registerAlertSourceForm: any;
+    showRegisterAlertSourceForm: boolean = false;
+    v2cFields: boolean = false;
+    v3Fields: boolean = false;
+    volumesArr = [];
+    allVolumes: any = [];
+    poolsArr = [];
+    allPools: any = [];
+    dataSource: any = [];
+    totalRecords: number;
+    loading: boolean;
+    storageOverview;
+    volumeOverview;
+    poolOverview;
+    groupedByVendor;
+    totalArrayRawCapacity: any;
+    totalArrayUsableCapacity: any;
+    totalArrayFreeCapacity: any;
+    allActiveAlerts: any = [];
+    scrollerData: any = [];
+    msgs: Message[];
+
+    label = {
+        name: this.i18n.keyID["sds_block_volume_name"],
+        description: this.i18n.keyID["sds_block_volume_descri"],
+        vendor: "Vendor",
+        model: "Model",
+        status: "Status",
+        host: "Host IP",
+        port: "Port",
+        username: "Username",
+        password: "Password",
```

Comment:
Why password in the description?

Reply:
This is a remnant of old code. Will fix this. 
I had written Register storage as a pop-up modal and for that this object held the labels for the form. This `password` displays the label of the form .
Since then Register Storage is moved to a new page.

Reply:
Is it fixed or will be fixed in new PR?

Reply:
This is pushed in the latest commit. Fixed.